Sources/Build/Command.compile(ClangModule).swift

```diff
@@ -61,12 +61,12 @@ struct ClangModuleBuildMetadata {
         return module.recursiveDependencies.flatMap { module in
             switch module {
             case let module as ClangModule:
-                 let product = Product(name: module.name, type: .Library(.Dynamic), modules: [module])
-                return product.targetName
-            case let module as CModule:
-                return module.targetName
+                let product = Product(name: module.name, type: .Library(.Dynamic), modules: [module])
+                return prefix.appending(product.outname).asString
+            case is CModule:
+                return nil
             case let module as SwiftModule:
-                return module.targetName
+                return prefix.appending(component: module.c99name + ".swiftmodule").asString
```

Comment:
The `module.c99name + ".swiftmodule"` bit could be factored out into a module property.  This entire logic is in need of a rewrite to make it more layered, however, so I am not sure how important this is to fix in this PR.